packages/core/src/lib/vector-input.svelte

```diff
@@ -0,0 +1,82 @@
+<!--
+@component
+  
+For numeric user inputs that require easy adjustment.
+
+```svelte
+    <VectorInput
+    type="number"
+    step={1}
+    labels={['x', 'y', 'z']}
+    placeholders={{
+      x: '0',
+      y: '0',
+      z: '0'
+    }}
+    values={{
+      x: 0,
+      y: 0,
+      z: 0
+    }}
+  />
+```
+-->
+<svelte:options immutable />
+
+
+<script lang='ts'>
```

Comment:
Spacing is a little funny in this script here - do you mind breaking it up a little bit?